src/ds/thread_messaging.h

```diff
@@ -67,7 +69,14 @@ namespace threading
       ThreadMsg* current = local_msg;
       local_msg = local_msg->next;
 
-      current->cb(std::unique_ptr<ThreadMsg>(current));
+      if (current->cb)
+        current->cb(std::unique_ptr<ThreadMsg>(current));
+      else
+      {
+        current->f();
```

Comment:
Can we give `f` a better name? It looks like its a fire-once-then-kill-me callback or something?

Reply:
`f` is the `f`unction. I just wanted to use lambdas for thread functions and add a way to make them automatically retry upon failure. I might remove all of that though.